### PR TITLE
android-studio: add e2fsprogs dependency

### LINKED ISSUE #171355: android-studio: Unable to load library 'e2p'
### Describe the bug
Starting `android-studio` results in:

```
2022-05-03 04:47:25,050 [   3829]   WARN - openapi.util.io.FileSystemUtil - Unable to load library 'e2p':
libe2p.so: cannot open shared object file: No such file or directory
libe2p.so: cannot open shared object file: No such file or directory
Native library (linux-x86-64/libe2p.so) not found in resource path (/nix/store/9pdlcs9jjf7n9x249ybzzcg3v0kjnqzy-android-studio-stable-2021.1.1.23-unwrapped/lib/bootstrap.jar:/nix/store/9pdlcs9jjf7n9x249ybzzcg3v0kjnqzy-android-studio-stable-2021.1.1.23-unwrapped/lib/util.jar:/nix/store/9pdlcs9jjf7n9x249ybzzcg3v0kjnqzy-android-studio-stable-2021.1.1.23-unwrapped/lib/jdom.jar:/nix/store/9pdlcs9jjf7n9x249ybzzcg3v0kjnqzy-android-studio-stable-2021.1.1.23-unwrapped/lib/log4j.jar:/nix/store/9pdlcs9jjf7n9x249ybzzcg3v0kjnqzy-android-studio-stable-2021.1.1.23-unwrapped/lib/jna.jar:/nix/store/9pdlcs9jjf7n9x249ybzzcg3v0kjnqzy-android-studio-stable-2021.1.1.23-unwrapped/jre/lib/tools.jar) 
java.lang.UnsatisfiedLinkError: Unable to load library 'e2p':
libe2p.so: cannot open shared object file: No such file or directory
libe2p.so: cannot open shared object file: No such file or directory
Native library (linux-x86-64/libe2p.so) not found in resource path (/nix/store/9pdlcs9jjf7n9x249ybzzcg3v0kjnqzy-android-studio-stable-2021.1.1.23-unwrapped/lib/bootstrap.jar:/nix/store/9pdlcs9jjf7n9x249ybzzcg3v0kjnqzy-android-studio-stable-2021.1.1.23-unwrapped/lib/util.jar:/nix/store/9pdlcs9jjf7n9x249ybzzcg3v0kjnqzy-android-studio-stable-2021.1.1.23-unwrapped/lib/jdom.jar:/nix/store/9pdlcs9jjf7n9x249ybzzcg3v0kjnqzy-android-studio-stable-2021.1.1.23-unwrapped/lib/log4j.jar:/nix/store/9pdlcs9jjf7n9x249ybzzcg3v0kjnqzy-android-studio-stable-2021.1.1.23-unwrapped/lib/jna.jar:/nix/store/9pdlcs9jjf7n9x249ybzzcg3v0kjnqzy-android-studio-stable-2021.1.1.23-unwrapped/jre/lib/tools.jar)
	at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:301)
	at com.sun.jna.NativeLibrary.getInstance(NativeLibrary.java:461)
...
```

### Steps To Reproduce
1. Start `android-studio` from a terminal

### Expected behavior
No such error

### Screenshots
N/A

### Additional context
Upstream JetBrains editors added `libe2p.so` as a dependency, and was already fixed for common JetBrains editors. See:
https://github.com/NixOS/nixpkgs/issues/119618
and the fix:
https://github.com/NixOS/nixpkgs/pull/134616

Fixing this issue should be a simple matter of adding the `e2fsprogs` dependency.

### Notify maintainers
I'm not sure who is maintaining this? The derivation just says `with maintainers:` and I'm not sure from where it's coming from.
Based on Git history: @alapshin 

### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.34, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.7.0`
 - channels(ari): `""`
 - channels(root): `"home-manager-21.11, nixos-21.11.337114.9887f024766"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by ari-becker**
Closes #171355

###### Description of changes

Adds the `e2fsprogs` dependency

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by alapshin** [comment]
Looks good to me.

**Comment by ari-becker** [comment]
It seems like @GrahamcOfBorg is refusing to evaluate because `android-studio` has a non-free license: https://logs.nix.ci/?attempt_id=cbf654d8-c811-4d85-94af-88f04feea553&key=nixos%2Fnixpkgs.171356

I don't have write access to the repository to merge this. How can we proceed?
@alapshin ? (tagged because maintainer)
@grahamc ? (tagged because ofborg doesn't make it clear how to proceed)

**Review by alapshin — APPROVED** [review]

**Comment by alapshin** [comment]
I don't have write access either. We will have to wait for someone to merge it. Regarding borg I'm not sure how to proceed. I don't remember having such problem in the past.

**Pull request merged**
